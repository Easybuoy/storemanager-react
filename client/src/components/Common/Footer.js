import React from 'react'

const Footer = () => {
  return (
    <div>
      <footer id="footer" className="footerstickbottom">
            <p>Store Manager Copyright &copy; 2018</p>
    </footer>
    </div>
  )
}

export default Footer;