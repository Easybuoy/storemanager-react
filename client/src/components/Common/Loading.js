import React from 'react'
import { Triple } from 'react-preloading-component';

const Loading = () => {
  return (
    <div style={{
        position: 'fixed',
        top: '50%',
        left: '50%',
        marginTop: '-9em',
        marginLeft: '-15em',
        width: '30em',
        height: '18em'
    }}>
      <Triple color="#0099ff"/>
    </div>
  )
}

export default Loading;