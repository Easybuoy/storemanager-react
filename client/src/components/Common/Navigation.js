
import React, { Component } from 'react'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { signOut } from '../../actions/authActions';

class Navigation extends Component {
    openSlideMenu(){
        document.getElementById('side-menu').style.width = '250px';
    }

    closeSlideMenu(){
        document.getElementById('side-menu').style.width = '0';
    }

  render() {
    let userProfilePicture = this.props.auth.user.userImage;
    userProfilePicture = `http://store--manager.herokuapp.com/${userProfilePicture}`;
    let dashboardLinks = (
        <div id="side-menu" className="side-nav">
        <a href="#" className="btn-close" onClick={this.closeSlideMenu}>&times;</a>
            <li className="current"><Link to="/dashboard">Dashboard</Link></li>
            <li><Link to="/viewsales">View Sales Record</Link></li>
            <li><Link to="/profile">Profile</Link></li> 
            <li><a onClick={this.props.signOut}>Logout</a></li>
        </div>
    );
    if (this.props.auth.user.type === 1) {
        dashboardLinks = (
            <div id="side-menu" className="side-nav">
            <a href="#" className="btn-close" onClick={this.closeSlideMenu}>&times;</a>
            <li className="current"><Link to="/dashboard">Dashboard</Link></li>
            <li><Link to="/createproduct">Create Product</Link></li>
            <li><Link to="/viewproducts">View Products</Link></li>
            <li><Link to="/createattendant">Create Sales Attendant</Link></li>
            <li><Link to="/viewsales">View Sales</Link></li>
            <li><Link to="/viewattendants">View Attendants</Link></li>
            <li><a onClick={this.props.signOut}>Logout</a></li>
        </div>
        );
    }
    return (
      <div>
         <nav className="navbar">
            <span className="open-slide">
                <a href="#" id="openslidemenu" onClick={this.openSlideMenu}>
                <svg width="30" height="30">
                        <path d="M0,5 30,5" stroke="#ffffff"
                        strokeWidth="5"/>
                        <path d="M0,14 30,14" stroke="#ffffff"
                        strokeWidth="5"/>
                        <path d="M0,23 30,23" stroke="#ffffff"
                        strokeWidth="5"/>
                </svg>
            </a></span>

            <ul>
                <li><a href="store_attendant_profile.html" style={{ paddingTop : '10px' }}><div><img id="userimg" src={userProfilePicture} /> </div></a></li>
                <li id="shoppingcart"><a href="#"><label  id="shoppingcartlabel">0</label ><img id="cartimage" src="http://store--manager.herokuapp.com/img/cart.png" /></a></li>
                    <li><a href="#"><div><input className="search" placeholder="Search Products" type="text" /> <i id="searchimg" className="fas fa-search fa-2x"></i> </div></a></li>
                    <li id="appnameli"><a href="dashboard.html"><h1 id="appname"><span className="highlight">Store</span> Manager</h1></a></li>
            </ul>
        </nav>

        
        {dashboardLinks}
      </div>
    )
  }
}

Navigation.propTypes = {
    signOut: PropTypes.func.isRequired,
    auth: PropTypes.object.isRequired,
}

const mapStateToProps = state => ({
    auth: state.auth
});

export default connect(mapStateToProps, { signOut })(Navigation);