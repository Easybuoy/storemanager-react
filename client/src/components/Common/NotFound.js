import React from 'react'
import { Link } from 'react-router-dom';

import NotFoundImage from '../../../assets/images/page-not-found.png';

const NotFound = () => {
  return (
    <div className="text-center">
    <img src={ NotFoundImage } className="notfound"/>
        <div className="text-center">
        <Link to="/dashboard"><button className="button_1 text-center">Back To Dashboard</button></Link>
        </div>
          </div>
  )
}

export default NotFound;