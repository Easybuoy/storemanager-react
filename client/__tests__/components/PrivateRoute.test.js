import React from 'react';
import { shallow } from 'enzyme';

import { PrivateRoute } from '../../src/components/Common/PrivateRoute';

describe('<PrivateRoute />', () => {
    const props = {
        auth: jest.mock(),
      };

    it('renders the PrivateRoute component correctly', () => {
        const wrapper = shallow(<PrivateRoute {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render component if auth.isSignedIn = true', () => {
        props.auth.isSignedIn = true;
        const wrapper = shallow(<PrivateRoute {...props} />)

        expect(wrapper).toMatchSnapshot();
        });
});