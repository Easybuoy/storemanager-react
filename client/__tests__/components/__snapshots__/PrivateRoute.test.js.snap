// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> renders the PrivateRoute component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrivateRoute
    auth={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<PrivateRoute /> should render component if auth.isSignedIn = true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrivateRoute
    auth={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "isSignedIn": true,
        "mock": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
